### PR TITLE
Allow extensions to use the backing store

### PR DESCRIPTION
**Pull request by theopolis**
This added test ensures extension processes can set/get database values.

### CODE REVIEW

**Comment by osqueryer** [comment]
Build origin/pr/1907/merge (Job: 2447) failed :-1:

**Comment by osqueryer** [comment]
Build origin/pr/1907/merge (Job: 2448) is good! :+1:

File: osquery/examples/example_extension.cpp [hunk 1]
```diff
@@ -43,8 +43,48 @@ class ExampleTable : public TablePlugin {
   }
 };
 
+/**
+ * @brief A more 'complex' example table is provide to assist with tests.
```
Comment (theopolis) [inline osquery/examples/example_extension.cpp]: Correct these grammar issues.

**Comment by osqueryer** [comment]
Build origin/pr/1907/merge (Job: 2450) is good! :+1:

**Pull request merged**